### PR TITLE
Complete decompilation of main.c

### PR DESCRIPTION
**Pull request by Cleverking2003**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -192,7 +345,7 @@ void sub_08152178(void) {
     CpuCopy32(gUnk_03002E70, (void*)REG_ADDR_WIN0H, 0xc);
     CpuCopy16(gUnk_030024E8, (void*)REG_ADDR_BLDCNT, 6);
     CpuCopy16(gUnk_03003680, (void*)REG_ADDR_BG0HOFS, 0x10);
-    CpuCopy32(gUnk_03002520, (void*)REG_ADDR_BG2PA, 0x20);
+    CpuCopy32(&gUnk_03002520, (void*)REG_ADDR_BG2PA, 0x20);
```
Comment (jiangzhengwenjz) [inline src/main.c:348]: `sizeof`

File: src/main.c [hunk 2]
```diff
@@ -94,7 +247,7 @@ void sub_08151DC4(void) {
     DmaCopy32(3, gUnk_03002E70, (void*)REG_ADDR_WIN0H, 0xc);
     DmaCopy16(3, gUnk_030024E8, (void*)REG_ADDR_BLDCNT, 6);
     DmaCopy16(3, gUnk_03003680, (void*)REG_ADDR_BG0HOFS, 0x10);
-    DmaCopy32(3, gUnk_03002520, (void*)REG_ADDR_BG2PA, 0x20);
+    DmaCopy32(3, &gUnk_03002520, (void*)REG_ADDR_BG2PA, 0x20);
```
Comment (jiangzhengwenjz) [inline src/main.c:250]: `sizeof`

File: src/main.c [hunk 3]
```diff
@@ -7,6 +7,159 @@
...
+    DmaFill32(3, 0, gUnk_03003A10, 0x10);
+    DmaWait(3);
+    gUnk_03006070 = 0;
+    gUnk_03002548 = 0;
+    DmaFill32(3, 0, gUnk_03002470, 0x10);
+    DmaWait(3);
+    DmaFill32(3, 0, gUnk_030068C0, 0x10);
+    DmaWait(3);
+    m4aSoundInit();
+    m4aSoundMode(0x0095FA00);
```
Comment (jiangzhengwenjz) [inline src/main.c:133]: `SOUND_MODE_DA_BIT_8 | SOUND_MODE_FREQ_15768 | (15 << SOUND_MODE_MASVOL_SHIFT) | (10 << SOUND_MODE_MAXCHN_SHIFT)`

File: include/main.h [hunk 4]
```diff
@@ -26,52 +47,84 @@ struct Unk_03006CB0 {
...
-extern u16 gUnk_03002520[0x10];
+extern struct Unk_03002520 gUnk_03002520;
+extern u32 gUnk_03002540;
+extern u16 gUnk_03002544;
 extern u8 gUnk_03002548;
+extern u16 gUnk_0300254C;
+extern u8 gUnk_03002550;
 extern u32 gUnk_03002554;
 extern u8 gUnk_03002558;
 extern u8 gUnk_03002760[0x280];
```
Comment (jiangzhengwenjz) [inline include/main.h:73]: It's `extern u8 gUnk_03002760[2][0x280];`

Comment (Cleverking2003) [reply include/main.h:73]: Maybe, but we don't have confirmation yet

**Comment by Cleverking2003** [comment]
Fixed, except for gUnk_03002760 one

File: include/main.h (hunk 4 above)
Comment (Cleverking2003) [reply include/main.h]: Ok, it actually makes sense

**Pull request merged**